Client/Frontend/Browser/URLBarView.swift

```diff
@@ -747,7 +747,7 @@ extension URLBarView: AppStateDelegate {
         if toolbarIsShowing {
             let showShareButton = HomePageAccessors.isButtonInMenu(appState)
             homePageButton.hidden = showShareButton
-            shareButton.hidden = !showShareButton
+            shareButton.hidden = !showShareButton || inOverlayMode
```

Comment:
Nit: I would move this OR statement into the `showShareButton` definition since being in overlay mode is a factor in determining if it should show.